datafusion/core/src/execution/disk_manager.rs

```diff
@@ -88,15 +93,23 @@ impl DiskManager {
                     local_dirs
                 );
                 Ok(Arc::new(Self {
-                    local_dirs: Mutex::new(local_dirs),
+                    local_dirs: Mutex::new(Some(local_dirs)),
                 }))
             }
+            DiskManagerConfig::Disabled => Ok(Arc::new(Self {
+                local_dirs: Mutex::new(None),
+            })),
         }
     }
 
     /// Return a temporary file from a randomized choice in the configured locations
     pub fn create_tmp_file(&self) -> Result<NamedTempFile> {
-        let mut local_dirs = self.local_dirs.lock();
+        let mut guard = self.local_dirs.lock();
+        let local_dirs = guard.as_mut().ok_or_else(|| {
+            DataFusionError::ResourcesExhausted(
+                "Cannot spill to temporary file as DiskManager is disabled".to_string(),
```

Comment:
If a user sees this it may not be clear what is happening or what they can do about it.



What do you think about  changing the message to something like

```suggestion
                "Memory Exhausted (DiskManager is disabled, cannot spill to temporary file)".to_string(),
```

As a follow up (which I will do), I think it would be excellent to pass in a context string to create_tmp_file to make the error more specific.


```rust
    pub fn create_tmp_file(&self, context: &str) -> Result<NamedTempFile> {
```